plugins/jobs/src/js/pages/JobDetailPage.js

```diff
@@ -0,0 +1,193 @@
+import { DIALOGS } from "./JobDetailPageContainer";
+import JobRunHistoryTable from "./JobRunHistoryTable";
+
+class JobDetailPage extends mixin(TabsMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {
+      runHistory: "Run History",
+      configuration: "Configuration"
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift()
+    };
+  }
+
+  getNavigationTabs() {
+    return <ul className="menu-tabbed">{this.tabs_getUnroutedTabs()}</ul>;
+  }
+
+  getDestroyConfirmDialog() {
+    return null;
+  }
+
+  getPrettySchedule(job) {
+    const schedules = job.schedules;
+    if (schedules == null || schedules.length === 0) {
+      return null;
+    }
+
+    const schedule = schedules[0];
+    if (schedule.enabled) {
+      return prettycron.toString(schedule.cron);
+    }
+  }
+
+  getSubTitle(job) {
+    const nodes = [];
+    const scheduleText = this.getPrettySchedule(job);
+    let longestRunningTask = null;
+    const longestRunningActiveRun = job
+      .getActiveRuns()
+      .getLongestRunningActiveRun();
+
+    if (longestRunningActiveRun != null) {
+      longestRunningTask = longestRunningActiveRun
+        .getTasks()
+        .getLongestRunningTask();
+    }
+
+    if (longestRunningTask == null && scheduleText == null) {
+      return null;
+    }
+
+    if (scheduleText != null) {
+      nodes.push(
+        <p className="text-overflow flush" key="schedule-text">
+          <Icon
+            className="icon-margin-right"
+            key="schedule-icon"
+            color="grey"
+            id="repeat"
+            size="mini"
+          />
+          <span>Scheduled {StringUtil.lowercase(scheduleText)}</span>
+        </p>
+      );
+    }
+
+    if (longestRunningTask != null) {
+      const dateCompleted = longestRunningTask.getDateCompleted();
+
+      const status = TaskStates[longestRunningTask.getStatus()];
+      const statusClasses = classNames("job-details-header-status", {
+        "text-success":
+          status.stateTypes.includes("success") &&
+          !status.stateTypes.includes("failure"),
+        "text-danger": status.stateTypes.includes("failure")
+      });
+
+      let timePrefix = null;
+      const shouldSuppressRelativeTime = dateCompleted == null;
+
+      if (shouldSuppressRelativeTime) {
+        timePrefix = "for ";
+      }
+
+      nodes.push(
+        <span className={statusClasses} key="status-text">
+          {status.displayName}
+        </span>,
+        <span key="time-running">
+          <TimeAgo
+            prefix={timePrefix}
+            suppressSuffix={shouldSuppressRelativeTime}
+            time={longestRunningActiveRun.getDateCreated()}
+          />
+        </span>
+      );
+    }
+
+    return nodes;
+  }
+
+  renderConfigurationTabView(job) {
+    return <JobConfiguration job={job} />;
+  }
+
+  renderRunHistoryTabView(job) {
+    return <JobRunHistoryTable job={job} />;
+  }
+
+  getActions() {
```

Comment:
I have an impression that this here is breaking a lot of contracts that the menu will rely on. Do you think in your plans to move further you could integrate this work with:

https://github.com/dcos/dcos-ui/pull/3023

I would not like to wait for the full menu to be done in order for this to work.

Reply:
This will be done in https://jira.mesosphere.com/browse/DCOS-38483 but as explicitly stated it's not part of this ticket. I needed to cut the scope somehow.